pkg/logql/engine.go

```diff
@@ -222,13 +227,21 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 }
 
 func (q *query) Eval(ctx context.Context) (promql_parser.Value, error) {
-	queryTimeout := time.Minute * 5
-	userID, err := tenant.TenantID(ctx)
-	if err != nil {
-		level.Warn(q.logger).Log("msg", fmt.Sprintf("couldn't fetch tenantID to evaluate query timeout, using default value of %s", queryTimeout), "err", err)
-	} else {
-		queryTimeout = q.limits.QueryTimeout(userID)
+	queryTimeout := q.timeout
+	if q.timeout == 0 {
+		// Engine didn't set query timeout or user already migrated to new limits:query_timeout configuration.
+		// In this case, we can safely use the new limits timeout.
+		// If the limits timeout isn't configured either, use former 5 min timeout.
```

Comment:
I don't see this happening here since we are overwriting `queryTimeout` with `q.limits.QueryTimeout(userID)` and it would return `1m`, i.e. the default value for that user limit.

Reply:
agreed, it should be rare, but since `tenant.TenantID` might error, I think we're in better shape with that conditional clause. WDYT?

Reply:
We are expected to have the tenantID set in the context, which is also required by downstream code. I think we can return an error if we don't find it.

Reply:
we can, but wouldn't it be a breaking change?

Reply:
I don't think so since all of our store code needs a tenantID in the context to know which user it belongs to. Infact we would have already looked for it even before reaching here, for e.g while enforcing max query parallelism in QF and max query length in Queriers.

Reply:
right, but other things like the logcli also use this code. I don't have a strong opinion against removing this conditional but I feel safer keeping it.

That said if you think removing it is a good call just lmk and I'll remove it. Just wanted to double check things to make sure I'm not missing something :smile:

Reply:
My main concern here is that this comment is not matching with what code is doing here:
`If the limits timeout isn't configured either, use former 5 min timeout.`
If engine timeout is not set, we are using limits timeout, which defaults to `1m`.
We should either fix the comment or change the code to match the comment.

Reply:
damnnnnnn got it, thanks for the patience :smile: 

yep, we're definitely defaulting to 1m instead of 5m in such cases. I reverted to not fallback to the 5min then and rewrote that comment, WDYT?